pkg/domainservice/adapterInterfaces.go

```diff
@@ -53,6 +53,11 @@ type DoguToLoad struct {
 	Version           string
 }
 
+type MaintenanceMode interface {
+	Activate(content MaintenancePageModel) error
+	Deactivate() error
```

Comment:
Can you please add documentation for the errors here? The domain normally uses the specific error type to determine what to do next (mark blueprint as failed or retry reconciliation).